spectral_cube/lower_dimensional_structures.py

```diff
@@ -1093,6 +1120,51 @@ def __getattribute__(self, attrname):
 
 class VaryingResolutionOneDSpectrum(OneDSpectrum, MultiBeamMixinClass):
     def __new__(cls, value, beams=None, **kwargs):
+        out = super(VaryingResolutionOneDSpectrum, cls).__new__(cls, value, **kwargs)
         if beams is not None:
-            cls.beams = beams
-        return super(VaryingResolutionOneDSpectrum, cls).__new__(cls, value, **kwargs)
+            out.beams = beams
+        return out
+
+    def with_beams(self, beams):
+        '''
+        Attach a new beam object to the OneDSpectrum.
+
+        Parameters
+        ----------
+        beams : `~radio_beam.Beams`
+            A new beams object.
+        '''
+
+        meta = self.meta.copy()
+        meta['beams'] = beams
+
+        return self._new_spectrum_with(beams=beams, meta=meta)
+
+    def with_beam(self, beam):
+        '''
+        Re-define here so a single Beam cannot be passed.
+        '''
+        raise ValueError("For VaryingResolutionOneDSpectrum's, use with_beams "
+                         "instead of with_beam.")
```

Comment:
Right... there's not a way to "undefine" a method.  This implies that we should have a `BaseOneDSpectrum` object and have `OneDSpectrum` and `VaryingResolutionOneDSpectrum` inherit from it, like we do with Cubes.

Reply:
Yeah. So we'd have `LDO` -> `BaseOneDSpectrum` -> `OneDSpectrum`, `VRODS`. And `beam` would have to be defined in `Projection`, too. Despite the multiple levels of inheritance, it will probably still be easier to understand...